### PR TITLE
[NPU] Use the friendly name of the tensor instead of the legacy name of it

### PR DESCRIPTION
**Pull request by pereanub**
### Details:
 - *Use friendly name of the tensor instead of legacy name of it*
 - *...*

### Tickets:
 - *ticket-id*

### CODE REVIEW

File: src/plugins/intel_npu/src/backend/include/zero_executor.hpp [hunk 1]
```diff
@@ -67,10 +68,23 @@ class ZeroExecutor final : public IExecutor {
 
     ze_graph_handle_t _graph = nullptr;
     ze_graph_properties_t _props{};
+
     std::unordered_map<std::string, ArgumentDescriptor> _inputs_desc_map;
     std::unordered_map<std::string, ArgumentDescriptor> _outputs_desc_map;
 
     std::array<std::shared_ptr<CommandQueue>, stage::COUNT> _command_queues;
+
+    template <typename T, std::enable_if_t<NotSupportArgumentProperties3(T), bool> = true>
```
Comment (lmielick) [inline src/plugins/intel_npu/src/backend/include/zero_executor.hpp:77]: Let's be consistent with typename -- used only in in createGraph but not here

Comment (pereanub) [reply src/plugins/intel_npu/src/backend/include/zero_executor.hpp:77]: sorry, but I do not understand. Could you explain a little bit more, please?

Comment (pereanub) [reply src/plugins/intel_npu/src/backend/include/zero_executor.hpp:77]: done

**Review by razvanapetroaie — COMMENTED** [review]
Can't speak about dynamic shapes' flow, but the rest looks fine to me.

**Comment by ArtemySkrebkov** [comment]
@DariaMityagina 

Can you please review changes related to the dynamic shapes?

**Comment by DariaMityagina** [comment]
> @DariaMityagina
> 
> Can you please review changes related to the dynamic shapes?

Sure, will test the changes shortly.

File: src/plugins/intel_npu/src/backend/src/zero_infer_request.cpp [hunk 2]
```diff
@@ -117,7 +117,13 @@ ZeroInferRequest::ZeroInferRequest(const std::shared_ptr<ZeroInitStructsHolder>&
                                               executorInputDescriptors.at(shapeBufferName),
                                               shapeBufferName);
 
-            auto allocator = zeroMemory::HostMemAllocator(backendPtr);
+            ov::Allocator allocator;
+            if (properties.flags & ZE_DEVICE_PROPERTY_FLAG_INTEGRATED) {
+                allocator = zeroMemory::HostMemAllocator(backendPtr, ZE_HOST_MEM_ALLOC_FLAG_BIAS_WRITE_COMBINED);
+            } else {
+                allocator = zeroMemory::HostMemAllocator(backendPtr);
+            }
```
Comment (ArtemySkrebkov) [inline src/plugins/intel_npu/src/backend/src/zero_infer_request.cpp:125]:
```suggestion
            ov::Allocator allocator = [&] {
                if (properties.flags & ZE_DEVICE_PROPERTY_FLAG_INTEGRATED) {
                    return zeroMemory::HostMemAllocator(backendPtr, ZE_HOST_MEM_ALLOC_FLAG_BIAS_WRITE_COMBINED);
                }
                return zeroMemory::HostMemAllocator(backendPtr);
            }();
```

A little shorter and initializes allocator.

File: src/plugins/intel_npu/src/backend/src/zero_infer_request.cpp [hunk 3]
```diff
@@ -134,16 +140,25 @@ ZeroInferRequest::ZeroInferRequest(const std::shared_ptr<ZeroInitStructsHolder>&
...
-            check_level_zero_attributes_match(shapeDescriptor,
-                                              executorOutputDescriptors.at(shapeBufferName),
-                                              shapeBufferName);
+                check_level_zero_attributes_match(shapeDescriptor,
+                                                  executorOutputDescriptors.at(shapeBufferName),
+                                                  shapeBufferName);
 
-            auto allocator = zeroMemory::HostMemAllocator(backendPtr);
-            allocate_tensor(outputName, shapeDescriptor, TensorType::Shape, allocator);
+                ov::Allocator allocator;
```
Comment (ArtemySkrebkov) [inline src/plugins/intel_npu/src/backend/src/zero_infer_request.cpp:153]: Can you consider reusing a single allocator for the whole constructor?

**Review by ArtemySkrebkov — COMMENTED** [review]
My comments are about style, So I am good to approve 👍

**Review by DariaMityagina — APPROVED** [review]
LGTM 👍

File: src/plugins/intel_npu/src/backend/src/zero_infer_request.cpp (hunk 3 above)
Comment (pereanub) [reply src/plugins/intel_npu/src/backend/src/zero_infer_request.cpp]: Using only one for states and outputs but still need another one for inputs. It needs to be created with different flags for inputs.

File: src/plugins/intel_npu/src/backend/src/zero_infer_request.cpp (hunk 2 above)
Comment (pereanub) [reply src/plugins/intel_npu/src/backend/src/zero_infer_request.cpp]: I kept the old implementation here with the thought of reusing the allocator for both inputs and shape input tensors.

**Review by razvanapetroaie — APPROVED** [review]

**Review by ArtemySkrebkov — APPROVED** [review]

**Pull request merged**